hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OmTableInsightTask.java

```diff
@@ -0,0 +1,373 @@
+        initializeReplicatedSizeMap();
+    final Collection<String> taskTables = getTaskTables();
+    final Collection<String> sizeRelatedTables = getTablesToCalculateSize();
+
+    while (eventIterator.hasNext()) {
+      OMDBUpdateEvent<String, Object> omdbUpdateEvent = eventIterator.next();
+
+      if (!taskTables.contains(omdbUpdateEvent.getTable())) {
+        continue;
+      }
+
+      String countKey = getTableCountKeyFromTable(omdbUpdateEvent.getTable());
+      String unReplicatedSizeKey =
+          getUnReplicatedSizeKeyFromTable(omdbUpdateEvent.getTable());
+      String replicatedSizeKey =
+          getReplicatedSizeKeyFromTable(omdbUpdateEvent.getTable());
+
+      try {
+        switch (omdbUpdateEvent.getAction()) {
+        case PUT:
+          objectCountMap.computeIfPresent(countKey, (k, count) -> count + 1L);
+
+          // Compute unreplicated and replicated sizes for size-related tables
+          if (sizeRelatedTables.contains(omdbUpdateEvent.getTable()) &&
+              omdbUpdateEvent.getValue() instanceof OmKeyInfo) {
+            OmKeyInfo omKeyInfo = (OmKeyInfo) omdbUpdateEvent.getValue();
+            unreplicatedSizeCountMap.computeIfPresent(unReplicatedSizeKey,
+                (k, size) -> size + omKeyInfo.getDataSize());
+            replicatedSizeCountMap.computeIfPresent(replicatedSizeKey,
+                (k, size) -> size + omKeyInfo.getReplicatedSize());
+          }
+          break;
+
+        case DELETE:
+          if (omdbUpdateEvent.getValue() != null) {
+            objectCountMap.computeIfPresent(countKey,
+                (k, count) -> count > 0 ? count - 1L : 0L);
+
+            // Compute unreplicated and replicated sizes for size-related tables
+            if (sizeRelatedTables.contains(omdbUpdateEvent.getTable()) &&
+                omdbUpdateEvent.getValue() instanceof OmKeyInfo) {
+              OmKeyInfo omKeyInfo = (OmKeyInfo) omdbUpdateEvent.getValue();
+              unreplicatedSizeCountMap.computeIfPresent(unReplicatedSizeKey,
+                  (k, size) ->
+                      size > omKeyInfo.getDataSize() ?
+                          size - omKeyInfo.getDataSize() : 0L);
+              replicatedSizeCountMap.computeIfPresent(replicatedSizeKey,
+                  (k, size) ->
+                      size > omKeyInfo.getReplicatedSize() ?
+                          size - omKeyInfo.getReplicatedSize() : 0L);
+            }
+          }
+          break;
+
+        default:
+          LOG.trace("Skipping DB update event : Table: {}, Action: {}",
+              omdbUpdateEvent.getTable(), omdbUpdateEvent.getAction());
+        }
+      } catch (Exception e) {
+        LOG.error(
+            "Unexpected exception while processing the table {}, Action: {}",
+            omdbUpdateEvent.getTable(), omdbUpdateEvent.getAction(), e);
+        return new ImmutablePair<>(getTaskName(), false);
+      }
+    }
+
+    // Write count data to DB if the sizeCountMap is not empty
```

Comment:
Same here, pls correct comment for all below three if statements

Reply:
I have removed it as it is self-explanatory and doesn't require a comment.